src/serverlessPlugin.d.ts

```diff
@@ -99,6 +99,14 @@ export interface FullHttpEvent {
         minimum?: number
       }
     >
+    formDataParameters?: Record<
+      string,
+      {
+        required?: boolean
+        type: "string" | "number" | "boolean" | "integer" | "array" | "file"

```

Comment:
Would you also need an `arrayItemsType` here as well?

Reply:
(as with queryStringParameters below)

Reply:
Formdata field can only be string, blob or file

Reply:
@desaintvincent can you share where you found that info? I'm reading the [Swagger spec](https://swagger.io/docs/specification/2-0/describing-parameters/#form-parameters), and seeing
> That is, the operation’s consumes property must specify one of these content types. Form parameters are defined as in: formData. They can only be primitives (strings, numbers, booleans) or arrays of primitives (meaning you cannot use a $ref as the items value). Also, form parameters cannot coexist with the in: bodyparameter, because formData is a specific way of describing the body. To illustrate form parameters, consider an HTML POST form:

Reply:
On that note, though, @deepankarm that would require a bit more complicated logic, that is, we'd have to
1. force the user to specify a `consumes` property if they used `formDataParameters`
2. Error if the user uses both `formDataParameters` and `bodyType`

Reply:
Perhaps it would be easier to just piggy-back on `bodyType`, and just use `formDataParameters` if the user specifies a `consumes` type of `application/x-www-form-urlencoded` or `multipart/form-data`?

Reply:
@desaintvincent Do you see any issues with that solution? It sounds like `formDataParameters` are just an alternate form of specifying the body, so we could kill two birds with one stone by using `bodyType` _and_ `consumes` to acheive what @deepankarm is wanting from this library

Reply:
We should enable arguments for `consumes` & `produces` along with `formData` to support this. 

Apologies for the delay @bfaulk96. I've not got time to work on this PR after your latest merge to master. Will try to get something soon.

Reply:
No need to apologize for the delay! I just wanted to talk through potentially cleaner solutions

Reply:
Sorry about more conflicts... I'm really trying to make an effort to clean up the repo to make it easier for people to contribute in general

Reply:
The simpler, the better: Why not just allow a parameter to specify the header content-type, and let the usual bodyType? it's up to the dev to make both match (I was reading fast, and forgot the initial problem. Hope I'm not off topic x) )

ps: all src/ServerlessAutoSwagger.ts lines changed. It's hard to review and understand the context. Lint problem maybe?

Reply:
deepankarm hasn't updated the PR with the latest changes yet, which is probably why every line has changed.

Reply:
re:
> The simpler, the better: Why not just allow a parameter to specify the header content-type, and let the usual bodyType? it's up to the dev to make both match (I was reading fast, and forgot the initial problem. Hope I'm not off topic x) )

This is sort of what I was suggesting, although instead of specifying the header type (since that doesn't help so much in Swagger) you'd define the `consumes` so that we could potentially allow file upload in Swagger (if that's even possible)